mods/ra/maps/siberian-conflict-1-fresh-tracks/fresh-tracks.lua

```diff
@@ -0,0 +1,282 @@
+FinalConvoyPath = { Patrol7, Patrol6, Patrol5, FinalConvoyPath, ExitEast }
+
+OpeningMoves = function()
+	Utils.Do(Squad1, function(actor)
+		if not actor.IsDead then
+			actor.AttackMove(AlliedBase.Location)
+		end
+	end)
+
+	Utils.Do(Squad2, function(actor)
+		if not actor.IsDead then
+			actor.AttackMove(AlliedBase.Location)
+		end
+	end)
+
+	PatrolMammoth.Patrol(PatrolMammothPath, true, 10)
+end
+
+AttackWaveDelays =
+{
+	easy = { DateTime.Seconds(40), DateTime.Seconds(80) },
+	normal = { DateTime.Seconds(30), DateTime.Seconds(60) },
+	hard = { DateTime.Seconds(20), DateTime.Seconds(40) }
+}
+
+AttackWaves = function()
+	local attackpath = Utils.Random(AttackPaths)	
+	local attackers = Reinforcements.Reinforce(ussr, Utils.Random(AttackWaveUnits), { attackpath[1] })
+	Utils.Do(attackers, function(unit)
+		Trigger.OnAddedToWorld(unit, function()
+			unit.AttackMove(AlliedBase.Location)
+			unit.Hunt()
+		end)
+	end)
+
+	Trigger.AfterDelay(Utils.RandomInteger(AttackWaveDelay[1], AttackWaveDelay[2]), function()
+		AttackWaves()
+	end)
+end
+
+ConvoyWaves = 
+{
+	easy = 2,
+	normal = 3,
+	hard = 4
+}
+ConvoysPassed = 0
+
+SendConvoys = function()
+	ConvoysSent = true
+	Media.PlaySpeechNotification(allies, "ConvoyApproaching")
+	local path = Utils.Random(ConvoyPaths)
+	local units = Reinforcements.Reinforce(ussr, Utils.Random(ConvoyUnits), { path[1] })
+	local lastWaypoint = path[#path]
+	Utils.Do(units, function(unit)
+		Trigger.OnAddedToWorld(unit, function()
+			if unit.Type == "truk" then
+				Utils.Do(path, function(waypoint)
+					unit.Move(waypoint)
+				end)
+
+				Trigger.OnIdle(unit, function()
+					unit.Move(lastWaypoint)
+				end)
+			else
+				unit.Patrol(path)
+				Trigger.OnIdle(unit, function()
+					unit.AttackMove(lastWaypoint)
+				end)
+			end
+		end)
+	end)
+
+	ConvoysPassed = ConvoysPassed + 1
+	if ConvoysPassed <= ConvoyWaves[Map.LobbyOption("difficulty")] then
+		Trigger.AfterDelay(DateTime.Seconds(90), SendConvoys)
+			Media.PlaySpeechNotification(allies, "ConvoyApproaching")
+	else
+		FinalTrucks()
+	end
+end
+
+FinalTrucks = function()
+	Trigger.AfterDelay(DateTime.Minutes(1), function()
+		Media.PlaySpeechNotification(allies, "ConvoyApproaching")
+		local baseConvoy = Reinforcements.Reinforce(ussr, BaseConvoyUnits, { BaseEntry.Location })
+		Utils.Do(BaseConvoyPath, function(waypoint)
+			Utils.Do(baseConvoy, function(actor)
+				Trigger.OnAddedToWorld(actor, function(a)
+					a.Move(waypoint.Location)
+				end)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(2), function()
+		Media.PlaySpeechNotification(allies, "ConvoyApproaching")
+		local finalConvoy = Reinforcements.Reinforce(ussr, FinalConvoyUnits, { NorthEntry.Location })
+		Utils.Do(FinalConvoyPath, function(waypoint)
+			Utils.Do(finalConvoy, function(actor)
+				Trigger.OnAddedToWorld(actor, function(a)
+					a.Move(waypoint.Location)
+				end)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(127), function()
```

Comment:
Just the trucks need to be killed instead of the whole convoy. I guess I could split it up into a separate group of reinforcements that are just trucks. Also, it is theoretically possible, though silly, that a player could stop a truck from an earlier convoy with non-combatants, which is why I chose to `GetActors` here.

Reply:
I was thinking of something like
```
Trigger.OnAllKilled(Utils.Where(finalConvoy, function(a) return a.Type == "truk" end), function()
	Allies.MarkCompletedObjective(StopTrucks)
end)
```